semver.md

```diff
@@ -319,6 +319,20 @@ Abbreviating "version" as "v" is often seen with version control. Example:
 `git tag v1.2.3 -m "Release version 1.2.3"`, in which case "v1.2.3" is a tag
 name and the semantic version is "1.2.3".
 
+### Is there a regular expression (RegEx) to check a SemVer string?
+
+There are two. One with named groups for those systems that support them:
+
+```
+^(?<major>0|[1-9]\d*)\.(?<minor>0|[1-9]\d*)\.(?<patch>0|[1-9]\d*)(?:-(?<prerelease>(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+(?<buildmetadata>[0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$
+```
+
+And one with non-capturing groups instead (so cg1 = major, cg2 = minor,
```

Comment:
These both have capture groups, the second one has numbered capture groups, commonly referred to as just "capture groups", the first one has "named capture groups". A non-capturing regex would only be useful as a validity test, either the string matches or it doesn't.

Reply:
Well both have non-capturing groups as well. They are for groupings that are structurally necessary (e.g. in combination with qualifiers) but should not be referenceable as a numbered capture group. 

But you are correct that this description of the second regex is a bit misleading. I also haven't noticed this before. How about "And one where the numbered regex groups correspond to the segments of SemVer (so cg1 = major, ..."

Reply:
Would "with unnamed capture groups" or "with non-named capture groups" be more succinct?

Reply:
Would "And one with numbered capture groups instead" be more succinct?
The parenthesised information remains - I guess this assumes a tiny bit higher level of understanding of regexs, but do we otherwise start explaining capture groups?

Reply:
Yes, I think this is sufficient. The people who will want to know this, will most likely have a general understanding of RegExs and capture groups. Giving a short overview which capture group returns which segment of the SemVer is what they might want to know at hand. This allows the usage of the RegEx without fully understanding the details (non capturing groups, etc.).

Reply:
Is it really common that an implementation supports capture groups and not named capture groups?

Reply:
@jwdonahue from a quick Google it seems older versions of some languages do not support them (e.g Java < v7) and even some not so old (ECMAScript < 2018), given that older is relative to one's requirements (Java v7 was only released a gnats eye-blink ago in terms of longer lived applications), then I'd opine that yes, it is common 😃

Reply:
@wolf99, interesting. Well, I think the FAQ should refer to two regex forms for capture groups, named and otherwise, and then there should be a non-capturing version for simple validity testing. Not every step in a DevOps process cares a gnat about any of the fields, as long as the string is well formed.

Reply:
@jwdonahue well the non-named groups are just parentheses which are supported by pretty much every system except POSIX/GNU BRE/ERE according to [this site](https://www.regular-expressions.info/refcapture.html). One could argue for making one that doesn't use non-capturing groups, so the reference numbers will be different, but then again, according to the same site, this only effects ORACLE and XML. So I would just leave it as it is. In case some people raise this question at some point (which I highly doubt), we can further extend the FAQ. But leave it concise for now.